SalesforceSDK/Core/Net/HttpCall.cs

```diff
@@ -314,20 +323,24 @@ public static HttpCall CreatePost(string url, string requestBody)
                 }
             }
             HttpResponseMessage message;
+
             try
             {
                 message = await _webClient.SendAsync(req);
-                HandleMessageResponse(message);
             }
-            catch (Exception ex)
+            catch (HttpRequestException ex)
             {
-                _httpCallErrorException = ex;
+                _httpCallErrorException = new DeviceOfflineException("Request failed to send, most likely because we were offline", ex);
```

Comment:
This could be because of a malformed request too, yeah? In that case, we should throw a different kind of exception.